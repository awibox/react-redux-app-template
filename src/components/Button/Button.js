import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
// Styles
import styles from './Button.scss';

const Button = (props) => {
  const { icon, children, onClick } = props;
  return (
    <div className={classNames(styles.btn, { [styles.btnIcon]: icon })} onClick={onClick}>
      {icon && <span className={styles.icon}><i className={`fa ${icon}`}></i></span>}
      {children}
    </div>
  );
};

Button.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.string,
};

export default Button;
