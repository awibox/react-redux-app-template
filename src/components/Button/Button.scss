@import '../../styles/variables';

.btn {
  border-radius: 4px;
  box-sizing: border-box;
  position: relative;
  font-size: 14px;
  -webkit-appearance: none;
  outline: none;
  transition: all 0.1s;
  display: inline-block;
  margin-bottom: 0;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  white-space: nowrap;
  padding: 6px 16px;
  line-height: 1.52857143;
  user-select: none;
  background-color: rgba(61, 94, 97, .9);
  color: #fff;
  &:hover {
    background-color: rgba(61, 94, 97, .8);
  }
  &:active {
    background-color: rgba(61, 94, 97, 1);
  }
}
.btnIcon {
  padding-left: 40px;
}
.icon {
  position: absolute;
  top: -3px;
  left: 10px;
  font-size: 25px;
}
