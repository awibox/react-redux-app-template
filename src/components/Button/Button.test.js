import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const clickFn = jest.fn();
const testIcon = 'fa-github';

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    expect(component).toMatchSnapshot();
  });
  it('should call clickFn on button click', () => {
    const component = shallow(<Button onClick={clickFn}>Test button</Button>);
    component.find('div.btn').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
  it('should have icon', () => {
    const component = shallow(<Button onClick={clickFn} icon={testIcon}>Test button</Button>);
    expect(component.find(`.${testIcon}`)).toExist();
  });
});
