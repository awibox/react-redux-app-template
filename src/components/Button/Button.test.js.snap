// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  className="btn"
  onClick={[MockFunction]}
>
  Test button
</div>
`;
